### PR TITLE
connecting to existing connections UI

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Q -> Toolkit

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/37f902d7-9c70-4fb0-a04b-d42461a6101a

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/4fb886a8-6166-40ed-9f21-c51d1daa6278

## Toolkit -> Q

Uploading idc-exist-connection(Toolkit -> Q).mov…

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/711af1cc-4975-42d0-a616-86a5510c310b

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt [hunk 1]
```diff
@@ -106,30 +107,27 @@ class WebviewBrowser(val project: Project) : LoginBrowser(project, WebviewBrowse
...
-                    // TODO: telemetry
+            "selectConnection" -> {
+                val connId = jsonTree.get("connectionId").asText()
+                this.selectionSettings[connId]?.let { settings ->
+                    settings.onChange(settings.currentSelection)
                 }
             }
 
+            "loginBuilderId" -> {
+                loginBuilderId(CODEWHISPERER_SCOPES + Q_SCOPES)
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt:118]: Should we combine these scopes as Q_SCOPES now?

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt [hunk 2]
```diff
@@ -190,6 +189,31 @@ class WebviewBrowser(val project: Project) : LoginBrowser(project, WebviewBrowse
...
+                        // TODO: rewrite scope logic, it's short term solution only
+                        val scopes = CODEWHISPERER_SCOPES + Q_SCOPES
+                        AwsRegionProvider.getInstance()[conn.region]?.let { region ->
+                            loginIdC(conn.sessionName, conn.startUrl, region, scopes)
+                        }
+                    }
+                }
+            }
+
+            println(bearerCreds)
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt:212]: nit: should be removed?

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt (hunk 1 above)
Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt:118]: yes, we should, let's do it in another PR

**Comment by Will-ShaoHua** [comment]
1. patch item logo not rendered
2. remove profilename in the displayed name

item-logo-fixed2


item-logo-fixed

File: plugins/core/webview/src/q-ui/components/qOptions.vue [hunk 3]
```diff
@@ -74,8 +91,32 @@ export default defineComponent({
...
+            if (connection.startUrl === SONO_URL) {
+                return LoginIdentifier.BUILDER_ID
             }
+
+            return LoginIdentifier.ENTERPRISE_SSO
         },
+        // TODO: duplicates in toolkitOptions, should leverage model/LoginOption interface
+        connectionTypeDescription(connection: AwsBearerTokenConnection): string {
+            if (connection.startUrl === SONO_URL) {
+                return 'AWS Builder ID'
```
Comment (manodnyab) [inline plugins/core/webview/src/q-ui/components/qOptions.vue:111]: Is this displayed anywhere? Does it need to be localized?

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt [hunk 4]
```diff
@@ -188,6 +198,33 @@ class ToolkitWebviewBrowser(val project: Project) : LoginBrowser(project, Toolki
...
+                .associate {
+                    it.id to BearerConnectionSelectionSettings(it) { conn ->
+                        if (conn.isSono()) {
+                            loginBuilderId(CODECATALYST_SCOPES)
+                        } else {
+                            // TODO: rewrite scope logic, it's short term solution only
+                            val scopes = if (conn.isSono()) {
+                                CODECATALYST_SCOPES
+                            } else {
+                                listOf(IDENTITY_CENTER_ROLE_ACCESS_SCOPE)
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt:216]: Should we have CODECATALYST_SCOPES here too?

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt [hunk 5]
```diff
@@ -188,6 +198,33 @@ class ToolkitWebviewBrowser(val project: Project) : LoginBrowser(project, Toolki
...
+            // existing connections
+            val bearerCreds = ToolkitAuthManager.getInstance().listConnections()
+                .filterIsInstance<AwsBearerTokenConnection>()
+                .associate {
+                    it.id to BearerConnectionSelectionSettings(it) { conn ->
+                        if (conn.isSono()) {
+                            loginBuilderId(CODECATALYST_SCOPES)
+                        } else {
+                            // TODO: rewrite scope logic, it's short term solution only
+                            val scopes = if (conn.isSono()) {
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt:213]: Why do we check for isSono here? these are the Idc scopes right?

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt:213]: oh yes this is redundant check

File: plugins/core/webview/src/q-ui/components/qOptions.vue (hunk 3 above)
Comment (Will-ShaoHua) [reply plugins/core/webview/src/q-ui/components/qOptions.vue:111]: yea sg

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt (hunk 4 above)
Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt:216]: yea let me ask in our chat to confirm this

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt]: Mano I think I will keep it as is for now

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt]: basically Codecatalyst has to be enabled by explicitly click getStarted node or via the setup auth page

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt]: i just think of a case: if the idc account doesn't have Codecatalyst application access/enabled, their login will always fail

**Review by manodnyab — APPROVED** [review]

**Pull request merged**